File: 0225-implement-stack-using-queues/solution.py
class MyStack:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.queue = []
        

    def push(self, x: int) -> None:
        """
        Push element x onto stack.
        """
        self.queue.insert(0, x)
        

    def pop(self) -> int:
        """
        Removes the element on top of the stack and returns that element.
        """
        result = self.queue[0]
        del self.queue[0]
        return result
        

    def top(self) -> int:
        """
        Get the top element.
        """
        return self.queue[0]

    def empty(self) -> bool:
        """
        Returns whether the stack is empty.
        """
        return not bool(len(self.queue))


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: 0225-implement-stack-using-queues/solution.cpp
class MyStack {
private:
    vector<int> queue;
public:
    MyStack() 
    {   
    }
    
    void push(int x) 
    {
        queue.push_back(x);
    }
    
    int pop() 
    {
        if(empty() == true) return -1;
        int result =  queue[queue.size()-1];
        queue.pop_back();
        return result;
    }
    
    int top() 
    {
        return queue[queue.size()-1];
        
    }
    
    bool empty() 
    {
        return queue.size() == 0;   
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */
